z3/snap.py

```diff
@@ -346,11 +364,13 @@ def backup_full(self, snap_name=None, dry_run=False):
                 capture=True))
         self._cmd.pipe(
             "zfs send '{}'".format(z_snap.name),
-            self._compress(
-                self._pput_cmd(
-                    estimated=estimated_size,
-                    s3_prefix=self.s3_manager.s3_prefix,
-                    snap_name=z_snap.name)
+            self._encrypt(
```

Comment:
:+1: The order is good, compress then encrypt.

Reply:
@rciorba I don't think the order is correct. `self._encrypt` takes as an argument the command which should *follow* encryption. The argument to `self._encrypt` here is the compression command; hence compression is *following* encryption.